### PR TITLE
Upgrade assertj 3.20.2

### PR DESCRIPTION
**Pull request by schlosna**
**Goals (and why)**:
Assertj 3.20.0 & 3.20.1 had bad ABI compatibility breaks in https://github.com/assertj/assertj-core/pull/2173 (see
https://github.com/palantir/atlasdb/pull/5511 ) and was reverted in https://github.com/assertj/assertj-core/commit/9df530d4b375b42e7007ae6470604532d435a5cdA and released as assertj 3.20.2.

**Implementation Description (bullets)**:
This also picks up the fix https://github.com/assertj/assertj-core/pull/2265 for https://github.com/assertj/assertj-core/issue/2264 and switches back to `MapAssert#containsExactlyInAnyOrderEntriesOf` that https://github.com/palantir/atlasdb/pull/5511/ changed

**Testing (What was existing testing like?  What have you done to improve it?)**:

**Concerns (what feedback would you like?)**: 

**Where should we start reviewing?**: version.props

**Priority (whenever / two weeks / yesterday)**: ASAP as the ABI breaks are causing downstream issues

### CODE REVIEW

File: versions.props [hunk 1]
```diff
@@ -40,7 +40,7 @@ net.jcip:jcip-annotations = 1.0
 one.util:streamex = 0.7.3
 org.apache.commons:commons-pool2 = 2.9.0
 org.apache.commons:commons-lang3 = 3.12.0
-org.assertj:assertj-* = 3.20.1
+org.assertj:assertj-* = 3.20.2
```
Comment (schlosna) [inline versions.props:43]: we may have to cut a `com.palantir.safe-logging` release with the assertj ABI revert as https://github.com/palantir/safe-logging/pull/542 bumped a bunch of folks to assertj 3.20.0

Comment (schlosna) [reply versions.props:43]: yep:

```
java.lang.NoSuchMethodError: 'org.assertj.core.api.ThrowableAssert com.palantir.logsafe.testing.Assertions.assertThatThrownBy(org.assertj.core.api.ThrowableAssert$ThrowingCallable)'
	at com.palantir.logsafe.testing.Assertions.assertThatLoggableExceptionThrownBy(Assertions.java:58)
```

File: changelog/@unreleased/pr-5516.v2.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+type: fix
+fix:
+  description: "Upgrade assertj 3.20.2\n\n**Goals (and why)**:\nAssertj 3.20.0 & 3.20.1
```
Comment (Jolyon-S) [inline changelog/@unreleased/pr-5516.v2.yml:3]: Please fix the changelog. Perhaps:

```
Upgrade AssertJ to 3.20.2 to fix ABI compatibility breaks.
```

**Review by Jolyon-S — APPROVED** [review]
Approved but removed merge when ready so that you can fix your changelog first.

File: changelog/@unreleased/pr-5516.v2.yml (hunk 2 above)
Comment (schlosna) [reply changelog/@unreleased/pr-5516.v2.yml]: Done, thanks!

**Pull request merged**